core/server/api_container/server/startosis_engine/startosis_runner.go

```diff
@@ -28,15 +32,29 @@ func NewStartosisRunner(interpreter *StartosisInterpreter, validator *StartosisV
 		startosisInterpreter: interpreter,
 		startosisValidator:   validator,
 		startosisExecutor:    executor,
+		mutex:                &sync.Mutex{},
 	}
 }
 
 func (runner *StartosisRunner) Run(ctx context.Context, dryRun bool, parallelism int, packageId string, serializedStartosis string, serializedParams string) <-chan *kurtosis_core_rpc_api_bindings.StarlarkRunResponseLine {
+	runner.mutex.Lock()
+	startosis_warning.Initialize()
 	// TODO(gb): add metric tracking maybe?
 	starlarkRunResponseLines := make(chan *kurtosis_core_rpc_api_bindings.StarlarkRunResponseLine)
-
 	go func() {
-		defer close(starlarkRunResponseLines)
+		defer func() {
+			warnings := startosis_warning.GetContentFromWarningSet()
+
+			if len(warnings) > 0 {
+				for _, warning := range warnings {
+					// TODO: create a new binding_constructor for warning message
+					starlarkRunResponseLines <- binding_constructors.NewStarlarkRunResponseLineFromInstructionResult(warning)
```

Comment:
can we make this a typed thing that we return so that on other end we can tell between a warning and another message so that we can color it yellow!

Reply:
i will do that in the next PR. will add another method in binding_constructor for that @h4ck3rk3y